docs/explainers/TraversingTransactions.md

```diff
@@ -0,0 +1,42 @@
+2. `security_id` - This is best thought of as analgous to a certificate id in the old fashioned paper security world. Every issuance has a `security_id`, and this ID should be used anytime another transaction is acting on the shares issued in the issuance event identified with this `security_id`. **Here are a couple illustrations of where and how `security_id` is used**:
+   1. To exercise an option, the `PlanSecurityExercise` transaction refers to a `security_id`. This must be the the `security_id` of the `PlanSecurityIssuance` event used to issue the option being exercised.
+   2. To transfer shares of stock, the `StockTransfer` transaction's `security_id` field should contain the id of the `security_id` value provided in the `StockIssuance` event used to issue the original shares being transferred.
+3. `custom_id` - is a freetext field provided to store a certificate ID (or some sort of other, implementer-specific id for a given issuance of securities).
+4. `share_numbers_issued` - is meant to track specific share numbers that are part of an issuance in situations where shares are **not** fungible. For example, in a jurisdiction where issuing shares 300 - 500 is legall significant and different from issuing shares 700 - 900, you'd need to use this field to track the share numbers that are part of a given issuance.
+
+## Terminal Transactions
+
+A **terminal** transaction is one which none of the securities originally issued as part of an issuance event remain issued and oustanding to the same holder. In many cases, our OCF transaction events allow _part_ of an issuance to be extinguished - e.g. a transfer or repurchase in part - but some of the original issuance will remain outstanding, in which case this type of event is **not** a terminal transaction. For example, where 1,000 shares are issued to Bob and then Bob transfers 500 to Frank, leaving 500 of the original 1,000 issued to Bob. While this is not a "terminal transaction:, where an OCF transaction acts on an issuance only in part, you'll see an optional `balance_security_id` field on the operative transaction. If a `balance_security_id` field is present, it indicates a remainder from the original `security_id` (as illustrated in the preceding example with Bob and Frank) is still held by the original holder but now under a new `security_id`.
+
+Keep in mind, however, that, even where the event acts on an issuance only in-part and some of the original issuance remains outstanding in the name of the original holder, the OCF spec requires that you us a new issuance event for this remainder of the original issuance. This remainder will have its own `security_id` for the "remainder" that remains outstanding in the name of the original holder. At this point, the original `security_id` from the original issuance is no longer in use.
+
+![](../images/OCF%20Terminal%20Transaction.png)
```

Comment:
Your visual aid doesn't match the example; the transferred shares in the partial transfer case should be 500 but is 1000.